application/lib/atom_widget.dart

```diff
@@ -92,6 +92,30 @@ class _AtomWidgetState extends State<AtomWidget>
 
   @override
   Widget build(BuildContext context) {
+    bool atomlessLandmark = false;
+    bool directionlessLandmark = false;
+    bool duplicateLandmark = false;
+    final Set<String> directions = <String>{};
+    for (final Landmark landmark
+        in (widget.atom?['landmark'] as LandmarksPropertyValue?)?.value ??
+            <Landmark>[]) {
+      if (landmark.direction == '' || landmark.direction == null) {
+        directionlessLandmark = true;
+        if (landmark.atom == null) {
+          atomlessLandmark = true;
+        }
+        continue;
+      }
+      if (landmark.atom == null) {
+        atomlessLandmark = true;
+        continue;
+      }
+      if (directions.contains(landmark.direction)) {
+        duplicateLandmark = true;
+        continue;
+      }
+      directions.add(landmark.direction!);
+    }
```

Comment:
we should cache this and update it when the atom changes (i'm pretty sure if you listen to the atom it'll tell you when it changes)